database/migrations/0017-combine_user_tables.ts

```diff
@@ -0,0 +1,154 @@
+/* MIGRATION TO COMBINE LOGIN_USERS WITH STATE_USERS TABLE
+ *
+ * This migration combines the table `login_users` with
+ * the `state_users` table, where the later is the target.
+ */
+
+export async function upgrade(queryFn: (query: string, values?: any[]) => Promise<Array<any>>) {
+  // Drop column `group_id` since it contains uniform data which is not the same as the uniform data
+  // on login_users. Since we do not need this data anyway, we sjust throw it away.
+  await queryFn('ALTER TABLE `state_users` DROP COLUMN `group_id`;')
+
+  // Remove the unique constraint from the pubkey
+  // TODO - check for code impact
+  await queryFn('ALTER TABLE `state_users` DROP INDEX `public_key`;')
+
+  // Allow NULL on the `state_users` pubkey like it is allowed on `login_users`
+  // TODO remove the random key shenanigans
+  await queryFn('ALTER TABLE `state_users` MODIFY COLUMN `public_key` binary(32) DEFAULT NULL;')
+
+  // instead use a unique constraint for the email like on `login_users`
```

Comment:
Sure we want to remove the unique constraint?

Reply:
```
// instead use a unique constraint for the email like on `login_users`
  // therefore do not allow null on `email` anymore
  await queryFn(
    'ALTER TABLE `state_users` MODIFY COLUMN `email` varchar(255) COLLATE utf8mb4_unicode_ci NOT NULL;',
  )
  // TODO - check for code impact
  await queryFn('ALTER TABLE `state_users` ADD CONSTRAINT `email` UNIQUE KEY (`email`);')
  ```
  
  It actually adds an unique constraint, not removes it. It disallows NULL on EMail to enable an unique constraint